### PR TITLE
test_models: run on route list

### LINKED ISSUE #25397: CI: run test car models on big route list
For the first version of this:
2. make selfdrive/test/test_models.py accept a route list
3. run test_models on a small route list (in jenkins, too slow for Actions)
4.  generate a route list with the goal of one seg per (VIN, dongle ID) combination
     * note some VINs won't be populated as the VIN query doesn't work, e.g. Japanese cars
     * use MKV segments

Eventually, we'll want infra to automatically pull segments for every (VIN, dongle ID) and update the route list. I'd also like to run the test on whole routes instead of just segments.

### PR DESCRIPTION
**Pull request by sshane**
Includes one segment from every dongle id.

Catching some bugs:

```python
FAILED run_test_models.py::TestCarModel_11_NISSAN_LEAF_2018::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'standstill': 14}
FAILED run_test_models.py::TestCarModel_26_HYUNDAI_SONATA_2020::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'cruiseState': 2}
FAILED run_test_models.py::TestCarModel_100_TOYOTA_HIGHLANDER_2017::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'cruiseState': 1}
FAILED run_test_models.py::TestCarModel_174_KIA_EV6_2022::test_car_interface - AssertionError: 5798 != 0
FAILED run_test_models.py::TestCarModel_201_TOYOTA_PRIUS_2017::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'cruiseState': 3}
FAILED run_test_models.py::TestCarModel_283_NISSAN_ROGUE_2019::test_panda_safety_rx_valid - AssertionError: False is not true
FAILED run_test_models.py::TestCarModel_283_NISSAN_ROGUE_2019::test_car_interface - AssertionError: 348 != 0
FAILED run_test_models.py::TestCarModel_321_HONDA_CIVIC_2016::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'controlsAllowed': 1}
FAILED run_test_models.py::TestCarModel_620_TOYOTA_PRIUS_2017::test_car_interface - AssertionError: 1881 != 0
FAILED run_test_models.py::TestCarModel_620_TOYOTA_PRIUS_2017::test_panda_safety_rx_valid - AssertionError: False is not true
FAILED run_test_models.py::TestCarModel_633_HONDA_ODYSSEY_2018::test_car_interface - AssertionError: 5799 != 0
FAILED run_test_models.py::TestCarModel_633_HONDA_ODYSSEY_2018::test_panda_safety_carstate - AssertionError: 2 is not false : panda safety doesn't agree with openpilot: {'brakePressed': 1, 'mainOn': 544}
FAILED run_test_models.py::TestCarModel_633_HONDA_ODYSSEY_2018::test_panda_safety_rx_valid - AssertionError: False is not true
FAILED run_test_models.py::TestCarModel_350_KIA_EV6_2022::test_car_interface - AssertionError: 5800 != 0
FAILED run_test_models.py::TestCarModel_698_NISSAN_ALTIMA_2020::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'standstill': 10}
FAILED run_test_models.py::TestCarModel_575_TOYOTA_PRIUS_2017::test_car_interface - AssertionError: 19 != 0
FAILED run_test_models.py::TestCarModel_413_HYUNDAI_IONIQ_5_2022::test_car_interface - AssertionError: 5799 != 0
FAILED run_test_models.py::TestCarModel_742_TOYOTA_COROLLA_2017::test_car_interface - AssertionError: 17 != 0
FAILED run_test_models.py::TestCarModel_553_HONDA_PILOT_2017::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'controlsAllowed': 1}
FAILED run_test_models.py::TestCarModel_755_KIA_EV6_2022::test_car_interface - AssertionError: 5798 != 0
FAILED run_test_models.py::TestCarModel_755_KIA_EV6_2022::test_panda_safety_rx_valid - AssertionError: False is not true
FAILED run_test_models.py::TestCarModel_432_HYUNDAI_IONIQ_5_2022::test_car_interface - AssertionError: 5799 != 0
FAILED run_test_models.py::TestCarModel_906_LEXUS_RX_2016::test_panda_safety_carstate - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'cruiseState': 1}
FAILED run_test_models.py::TestCarModel_784_HYUNDAI_KONA_ELECTRIC_2019::test_car_interface - AssertionError: 5799 != 0
FAILED run_test_models.py::TestCarModel_784_HYUNDAI_KONA_ELECTRIC_2019::test_panda_safety_rx_valid - AssertionError: False is not true
```

### CODE REVIEW

**Comment by sshane** [comment]
How to run: 

```
INTERNAL_SEG_LIST="/home/batman/openpilot/selfdrive/car/tests/test_models_segs.txt" pytest -n32 ./test_models.py
```

**Comment by sshane** [comment]
What it looks like in jenkins: https://jenkins.comma.life/blue/organizations/jenkins/openpilot/detail/big-test-models-temp/3/pipeline

File: Jenkinsfile [hunk 1]
```diff
@@ -155,6 +155,7 @@ pipeline {
             sh "git submodule update --init --depth=1 --recursive"
             sh "scons --clean && scons --no-cache -j42"
             sh "scons --clean && scons --no-cache --random -j42"
+            sh "cd selfdrive/car/tests && INTERNAL_SEG_LIST='selfdrive/car/tests/test_models_segs.txt' FILEREADER_CACHE='1' pytest -n42 --dist=loadscope ./test_models.py"
```
Comment (adeebshihadeh) [inline Jenkinsfile:158]: filereader cache isn't doing anything here

File: selfdrive/car/tests/test_models_segs.txt [hunk 2]
```diff
@@ -0,0 +1,3888 @@
+# HONDA ODYSSEY 2018
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models_segs.txt:1]: this file should probably be in LFS (or even pushed to the azure CI bucket)

Comment (sshane) [reply selfdrive/car/tests/test_models_segs.txt:1]: was going to add to LFS. adding list of private routes to public Azure bucket doesn't make sense to me

File: Jenkinsfile (hunk 1 above)
Comment (sshane) [reply Jenkinsfile]: mounted the /tmp dir

File: Jenkinsfile [hunk 3]
```diff
@@ -143,18 +143,19 @@ pipeline {
...
+              args '--user=root -v /tmp/comma_download_cache:/tmp/comma_download_cache'
             }
           }
           steps {
             sh "git config --global --add safe.directory '*'"
             sh "git submodule update --init --depth=1 --recursive"
-            sh "scons --clean && scons --no-cache -j42"
-            sh "scons --clean && scons --no-cache --random -j42"
+            sh "scons --clean && scons -j42"
+//             sh "scons --clean && scons --no-cache --random -j42"
```
Comment (sshane) [inline Jenkinsfile:157]:
```suggestion
             sh "scons --clean && scons --no-cache --random -j42"
```

Comment (sshane) [inline Jenkinsfile:157]:
```suggestion
            sh "scons --clean && scons -j42"
            sh "scons --clean && scons --no-cache --random -j42"
```

File: Jenkinsfile [hunk 4]
```diff
@@ -143,18 +143,19 @@ pipeline {
...
               filename 'Dockerfile.openpilot_base'
-              args '--user=root'
+              args '--user=root -v /tmp/comma_download_cache:/tmp/comma_download_cache'
             }
           }
           steps {
             sh "git config --global --add safe.directory '*'"
             sh "git submodule update --init --depth=1 --recursive"
-            sh "scons --clean && scons --no-cache -j42"
+            sh "scons --clean && scons -j42"
```
Comment (sshane) [inline Jenkinsfile:156]:
```suggestion
            sh "scons --clean && scons --no-cache -j42"
```

File: Jenkinsfile [hunk 5]
```diff
@@ -143,18 +143,19 @@ pipeline {
...
-              args '--user=root'
+              args '--user=root -v /tmp/comma_download_cache:/tmp/comma_download_cache'
             }
           }
           steps {
             sh "git config --global --add safe.directory '*'"
             sh "git submodule update --init --depth=1 --recursive"
             sh "scons --clean && scons --no-cache -j42"
             sh "scons --clean && scons --no-cache --random -j42"
+            sh "INTERNAL_SEG_LIST='selfdrive/car/tests/test_models_segs.txt' FILEREADER_CACHE='1' pytest -n42 --dist=loadscope selfdrive/car/tests/test_models.py"
```
Comment (sshane) [inline Jenkinsfile:158]:
```suggestion
            sh "INTERNAL_SEG_LIST=selfdrive/car/tests/test_models_segs.txt FILEREADER_CACHE=1 pytest -n42 --dist=loadscope selfdrive/car/tests/test_models.py"
```

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**